_posts/2019-12-13-basic-rails-container.markdown

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title: "A Basic Rails Container"
+date: 2019-12-13 12:30:00
+categories: ["DevOps", "docker", "tutorials"]
+author: "rdormer"
+---
+
+Here at Ombulabs, we do a lot of work with older versions of Ruby and Rails.  Recently, I've been taking a look at Docker, the virtual container platform, to see how it might help us manage the often idiosyncratic setup that applications we work with might require.
+
+As I've noted [before](), setting up your own containers from scratch is usually a bad idea.  Wherever you can, start with one a [base image]() to save you a *lot* of time.  We'll start with the Ubuntu base image:
+
+```
+FROM ubuntu
+RUN apt-get update
+```
+
+It's a good idea to update the apt-get database when you build your container, since the base image is essentially frozen in time and you may want newer packages as they become available.  You might think it's not a big deal if you're using a container to run a legacy application like we are here, but keep in mind that packages also routinely receive security updates, and you'll definitely want those.
+
+Next, you'll want to install the dependencies that Rails requires to run.  This won't just be Ruby - it'll be everything required to build Ruby and any native extensions you end up using.  At the time of writing, the minimal set of dependencies you'll need to install are:
+
+`RUN apt-get -y install ruby ruby-dev make gcc git zlib1g-dev`
```

Comment:
There's a `ruby` image https://hub.docker.com/_/ruby to avoid all of this, maybe a note that they can use that too?

Reply:
@bronzdoc Ernesto and I discussed it, and that particular image doesn't really fit out specific needs.